doc/dev_guide/internals/box_protocol.rst

```diff
@@ -935,13 +941,121 @@ Available IPROTO_FEATURES are the following:
   MsgPack extension support. Clients that don't support this feature will receive
   error responses for :ref:`IPROTO_EVAL <box_protocol-eval>` and
   :ref:`IPROTO_CALL <box_protocol-call>` encoded to string error messages.
-- ``IPROTO_FEATURE_WATCHERS = 3`` -- remote watchers support: IPROTO_WATCH,
-  IPROTO_UNWATCH, and IPROTO_EVENT commands.
-
-.. // TODO: document remote watchers commands
+- ``IPROTO_FEATURE_WATCHERS = 3`` -- remote watchers support: :ref:`IPROTO_WATCH <box_protocol-watch>`,
+  :ref:`IPROTO_UNWATCH <box_protocol-unwatch>`, and :ref:`IPROTO_EVENT <box_protocol-event>` commands.
 
 IPROTO_ID requests can be processed without authentication.
 
+..  _box-protocol-watchers:
+
+Watchers
+--------
+
+The commands below support asynchronous server-client notifications signaled
+with :ref:`box.broadcast() <box-broadcast>`.
+Servers that support the new feature set the ``IPROTO_FEATURE_WATCHERS`` feature in reply to the ``IPROTO_ID`` command.
+When a connection is closed, all watchers registered for it are unregistered.
+
+The remote :ref:`watcher <box-watchers>` protocol works in the following way:
+
+#.  The client sends an ``IPROTO_WATCH`` packet to subscribe to the updates of a specified key defined on the server.
+
+#.  The server sends an ``IPROTO_EVENT`` packet to the subscribed client after registration.
+    The packet contains the key name and its current value.
+    After that, the packet is sent every time the key value is updated with
+    ``box.broadcast()`` provided that the last notification was acknowledged (see below).
+
+#.  After receiving a notification, the client sends an ``IPROTO_WATCH`` packet to acknowledge the notification.
+
+#.  If the client doesn't want to receive any more notifications, it unsubscribes by sending
+    an ``IPROTO_UNWATCH`` packet.
+
+All the three request types are asynchronous -- a receiving end doesn't send a packet in reply to any of them.
+Therefore, neither of them has a sync number.
+
+..  _box_protocol-watch:
+
+IPROTO_WATCH = 0x4a
```

Comment:
`IPROTO_WATCH` and `IPROTO_UNWATCH` should probably be listed in the [Requests](https://www.tarantool.io/en/doc/latest/dev_guide/internals/box_protocol/#requests) sub-section, because they are user requests. The description could just contain a reference to the Watchers sub-section, like [`IPROTO_AUTH`](https://www.tarantool.io/en/doc/latest/dev_guide/internals/box_protocol/#iproto-auth-0x07) refers to the Authentication sub-section.